python/flask/access-request-in-wrong-handler.yaml

```diff
@@ -0,0 +1,43 @@
+rules:
+  - id: avoid-accessing-request-in-wrong-handler
+    patterns:
+      - pattern-either:
+        - pattern: |
+            @app.route(..., method="GET")
+            def $X(...):
+              $Y = flask.request.json
+              ...
+              return
```

Comment:
why do you want to match on `return` (also I think you want `return ...`)? Would you rather have:

```
def $X(...):
    ...
    $Y = flask.request.json
    ...
```

Reply:
I believe that with 
```
    $Y = flask.request.json
    ...
```
the pattern was matching the entire python file. return was like a stop canary.

Reply:
This is addressed with the following comment